src/main/kotlin/racing/RacingSimulator.kt

```diff
@@ -0,0 +1,34 @@
+package racing
+
+object RacingSimulator {
+    fun interactiveSimulate() {
+        val (carNumber, attemptNumber) = RacingScanner.inputRacing()
+        val result = virtualSimulate(carNumber, attemptNumber)
+        RacingPrinter.printSimulation(result)
+    }
+
+    fun virtualSimulate(carNumber: Int, attemptNumber: Int): List<List<Int>> {
+        require(carNumber >= 0) { "자동차 수는 음수가 될 수 없습니다." }
+        require(attemptNumber >= 0) { "시도 횟수는 음수가 될 수 없습니다." }
+
+        val cars = List(carNumber) { Car() }
+        val result = mutableListOf<List<Int>>()
+
+        repeat(attemptNumber) {
+            attemptMoveCars(cars)
+            result.add(getCarsPosition(cars))
+        }
```

Comment:
미리 List 생성과 repeat 함수를 사용하는 대신 List 생성자를 활용할 수 있을 것 같습니다. 😁 
```suggestion
      val result = (1..attemptNumber).map {
             attemptMoveCars(cars)
             getCarsPosition(cars)
       }
```

Reply:
오 훨씬 좋네요!!! 👍👍👍